### PR TITLE
Root parent version upgrade

### PR DESCRIPTION
**Pull request by Thumimku**

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java [hunk 1]
```diff
@@ -222,4 +227,40 @@ private boolean isFileInSubDirectory(File baseDir, File path) {
...
+    // Join the two paths together, then normalize so that any ".." elements
+    // in the userPath can remove parts of baseDirPath.
+    // (e.g. "/foo/bar/baz" + "../attack" -> "/foo/bar/attack")
+    final Path resolvedPath = baseDirPath.resolve(userPath).normalize();
+
+    // Make sure the resulting path is still within the required directory.
+    // (In the example above, "/foo/bar/attack" is not.)
+    if (!resolvedPath.startsWith(baseDirPath)) {
+            throw new IllegalArgumentException("User path escapes the base path");
+        }
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java:262]: Formatting issue.

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java [hunk 2]
```diff
@@ -222,4 +227,40 @@ private boolean isFileInSubDirectory(File baseDir, File path) {
...
+
+    return true;
+    }
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java:265]: Formatting issue.

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java [hunk 3]
```diff
@@ -92,13 +92,19 @@ public int getFailedDeployments() {
...
+            return null;
+        } else {
+            return new Date(lastDeployed.getTime());
+        }
     }
 
     public void setLastDeployed(Date lastDeployed) {
-
-        this.lastDeployed = lastDeployed;
+        if (lastDeployed == null) {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java:103]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/DeploymentRevision.java [hunk 4]
```diff
@@ -116,30 +119,36 @@ public void setFileHash(String fileHash) {
      * @return
      */
     public Date getDeployedDate() {
-
-        return deployedDate;
+        if (deployedDate == null) {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/DeploymentRevision.java:122]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java [hunk 5]
```diff
@@ -92,13 +92,19 @@ public int getFailedDeployments() {
     }
 
     public Date getLastDeployed() {
-
-        return lastDeployed;
+        if (lastDeployed == null) {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java:95]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/DeploymentRevision.java [hunk 6]
```diff
@@ -116,30 +119,36 @@ public void setFileHash(String fileHash) {
...
+        }
     }
 
     /**
      * @param deployedDate
      */
     public void setDeployedDate(Date deployedDate) {
-
-        this.deployedDate = deployedDate;
+        if (deployedDate == null) {
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/DeploymentRevision.java:133]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java [hunk 7]
```diff
@@ -49,18 +51,18 @@
     RemoteFetchConfiguration getRemoteFetchConfiguration(int fetchConfigurationId) throws RemoteFetchCoreException;
 
     /**
-     * @param tenant_id
+     * @param tenantId
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfigurationService.java:54]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/20/files

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchCoreConfiguration.java [hunk 8]
```diff
@@ -20,6 +20,9 @@
 
 import java.io.File;
 
+/**
+ * Holds core configuration
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchCoreConfiguration.java:24]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/20/files#diff-adf5cad5e65dd39351fefc62319a23f4R24

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/ValidationReport.java [hunk 9]
```diff
@@ -41,8 +45,32 @@ public void addMessage(String message) {
         this.validationMessages.add(message);
     }
 
-    public void addMessage(String format, Object... args) {
-        this.addMessage(new Formatter().format(format, args).toString());
+
+    public void addMessageForMandatoryValidation(String field) {
+        this.addMessage(new Formatter().format(ValidationFormat.MANDATORY_VALUE, field).toString());
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/ValidationReport.java:50]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/20/files#diff-a634874ab10cae14c8164a1cb955d861R55

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/actionlistener/ActionListenerBuilder.java [hunk 10]
```diff
@@ -63,6 +63,18 @@ public ActionListenerBuilder addConfigDeployer(ConfigDeployer configDeployer) {
         return this;
     }
 
+    public RemoteFetchConfiguration getFetchConfig() {
+        return fetchConfig;
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/actionlistener/ActionListenerBuilder.java:67]: Fixed in https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files#diff-47c19976b862160187e8eae17bf50e26R65

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/repomanager/RepositoryManagerBuilder.java [hunk 11]
```diff
@@ -57,6 +57,14 @@ public RepositoryManagerBuilder addRemoteFetchCoreConfig(RemoteFetchCoreConfigur
         return this;
     }
 
+    public RemoteFetchConfiguration getFetchConfig() {
+        return fetchConfig;
```
[the remark this answers is not in the document]
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/repomanager/RepositoryManagerBuilder.java:61]: https://github.com/wso2-extensions/identity-fetch-remote/pull/25/files#diff-47c19976b862160187e8eae17bf50e26R65

File: components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java (hunk 2 above)
Comment (Thumimku) [reply components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java:265]: Thank you, This comment was fixed in
https://github.com/wso2-extensions/identity-fetch-remote/pull/40
https://github.com/wso2-extensions/identity-fetch-remote/pull/41
https://github.com/wso2-extensions/identity-fetch-remote/pull/42
https://github.com/wso2-extensions/identity-fetch-remote/pull/43
https://github.com/wso2-extensions/identity-fetch-remote/pull/44
https://github.com/wso2-extensions/identity-fetch-remote/pull/45.
I resolve this conversation.

**Pull request merged**